File: src/main/java/dev/slimevr/vr/trackers/SensorTap.java
package dev.slimevr.vr.trackers;

public class SensorTap {

	public final boolean doubleTap;

	public SensorTap(int tapBits) {
		doubleTap = (tapBits & 0x40) > 0;
	}

	@Override
	public String toString() {
		return "Tap{" + (doubleTap ? "double" : "") + "}";
	}

	public enum TapAxis {
		X, Y, Z
	}
}

File: server/src/main/java/dev/slimevr/tracking/trackers/udp/SensorTap.kt
package dev.slimevr.tracking.trackers.udp

data class SensorTap(val tapBits: Int) {
	val doubleTap = tapBits and 0x40 > 0

	enum class TapAxis {
		X, Y, Z
	}
}

interface SensorSpecificPacket {
	val sensorId: Int
	companion object {
		/**
		 * Sensor with id 255 is "global" representing a whole device
		 *
		 * @param sensorId
		 * @return
		 */
		fun isGlobal(sensorId: Int): Boolean {
			return sensorId == 255
		}
	}
}
